src/main/resources/stubs/net/ip.gobra

```diff
@@ -110,9 +124,11 @@ func (ip IP) IsGlobalUnicast() bool /*{
 preserves forall i int :: 0 <= i && i < len(ip) ==> acc(&ip[i], 1/10000)
 ensures len(ip) == IPv4len ==> forall i int :: 0 <= i && i < len(ip) ==> &ip[i] == &res[i]
 ensures len(ip) == IPv6len && isZeros(ip[0:10]) && ip[10] == 255 && ip[11] == 255 ==> len(res) == IPv4len && forall i int :: 0 <= i && i < IPv4len ==> &ip[12+i] == &res[i]
+ensures (len(ip) == IPv6len && res != nil) ==> (len(res) == IPv4len && forall i int :: 0 <= i && i < IPv4len ==> &ip[12+i] == &res[i])
```

Comment:
Would this be equivalent to state on the line above that res won't be nil. However it seems weird that this is not already implied by `len(res) == IPv4len`. I bring this up because the RHS seems very similar

Reply:
I'm not sure if I understood this comment. Anyhow, I think I can further simplify this line from `ensures (len(ip) == IPv6len && res != nil) ==> (len(res) == IPv4len && forall i int :: 0 <= i && i < IPv4len ==> &ip[12+i] == &res[i])` to `ensures (len(ip) == IPv6len && res != nil) ==> (forall i int :: 0 <= i && i < IPv4len ==> &ip[12+i] == &res[i])` but I don't think I can simplify it further, because it might be the case that `res` is nil when the input has size `IPv6len`.

Reply:
I was basically looking at the implementation and it seems to me that the only case in which you get a non-nil result and `len(ip) == IPV6len` is the one covered by the specification in the line above.
The newly added postcondition only provides additional insights to clients that do not know whether `isZeros(ip[0:10]) && ip[10] == 255 && ip[11] == 255` holds but know that the result is non-nil

Reply:
Following this logic, you could save redundancy by changing line 126 to only imply that the result is non-nil (which would then enable clients to use the new implication on line 127

Reply:
I see, for some reason I didn't notice this. Thanks!